docs/FutureIdeas.md

```diff
@@ -66,3 +66,181 @@ and protocol witness tables, and leave all of this in a context object.
+```csharp
+    [SwiftNativeObjectTag()][SwiftTypeName("OverrideTests.MontyWSMBool")]
+    public class MontyWSMBool : SwiftNativeObject
+    {
+        static MontyWSMBool_xam_vtable xamVtableMontyWSMBool;
+        static MontyWSMBool()
+        {
+            XamSetVTable();
+        }
+        public static SwiftMetatype GetSwiftMetatype()
+        {
+            return NativeMethodsForXam_sub_MontyWSMBool.PIMetadataAccessor_Xam_sub_MontyWSMBool(SwiftMetadataRequest.Complete);
+        }
+        static IntPtr _XamMontyWSMBoolCtorImpl()
+        {
+        	// factory to construct the swift implementation and get back a handle
+        }
+        public  MontyWSMBool(): base(MontyWSMBool._XamMontyWSMBoolCtorImpl(), GetSwiftMetatype(), SwiftObjectRegistry.Registry)
+        {
+        }
+        protected  MontyWSMBool(IntPtr handle, SwiftMetatype mt, SwiftObjectRegistry registry): base(handle, mt, registry)
+        {
+        }
+        ~MontyWSMBool()
+        {
+            Dispose(false);
+        }
+        bool BaseVal()
+        {
+            IntPtr thisIntPtr = StructMarshal.RetainSwiftObject(this);
+            bool retval = default(bool);
+            // call into swift to call the base implementation of Val ()
+            retval = NativeMethodsForXam_sub_MontyWSMBool.PImethod_Xam_sub_MontyWSMBoolXamarin_xam_sub_MontyWSMBoolDxam_super_val(thisIntPtr);
+            SwiftCore.Release(thisIntPtr);
+            return retval;
+        }
+        public virtual bool Val()
+        {
+            return BaseVal(); // default implementation, aka, the base.Val
+        }
+#if __IOS__ || __MACOS__ || __TVOS__ || __WATCHOS__
+        [MonoPInvokeCallback(typeof(MontyWSMBool_xam_vtable.Delfunc0))]
+#endif
+		// gets called by swift to implement Val
+        static bool xamVtable_recv_Val(IntPtr self)
+        {
+            return SwiftObjectRegistry.Registry.CSObjectForSwiftObject <MontyWSMBool> (self).Val();
+        }
+        static void XamSetVTable()
+        {
+        }
+        struct MontyWSMBool_xam_vtable
+        {
+            public delegate bool Delfunc0(IntPtr self);
+            [MarshalAs(UnmanagedType.FunctionPtr)]
+            public Delfunc0 func0;
+        }
+    }
+```
+Now you can, in C# override this type and do change the behavior of `Val ()` and if you pass
+an instance of this C# type into Swift, it will pass the handle of the Swift override that
+vectors through your instance.
+
+There is one problem though - when you add new functionality to the the type, Swift will
+know nothing about it. If you have a dozen new C# subclasses of the original Swift class,
+you will not see the new functionality in Swift - they will all look like the singular
+Swift override.
+
+We can fix that - here's one way of doing that.
+Suppose we have an attribute named `[ExposeToSwift]` that can be applied to the type. That way if
+I write this in C#:
+```csharp
+[ExposeToSwift]
```

Comment:
For Objective-C this happens automatically for all types that subclass NSObject - for Swift we could also do something similar, where we expose all types that subclass a Swift type.